src/main/java/javaslang/control/Option.java

```diff
@@ -64,6 +65,20 @@
     }
 
     /**
+     *
+     * Wraps a Java Optional to a new Option
+     *
+     * @param optional a given optional to wrap in {@code Option}
+     * @param <T> type of the value
+     * @return {@code Some(optional.get())} if value is Java {@code Optional} is present, {@code None} otherwise
+     */
+    static <T> Option<T> wrap(Optional<T> optional) {
```

Comment:
Great, thx! I've two remarks:

* Let's take `Optional<? extends T>`, it is useful in some cases
* I would not call it _wrap_, because the `Optional` instance is not wrapped but converted somehow. `toOption` is already defined in `Value` and `asOption` would be confusing because we also have `toOption`. It is syntactic sugar/a shortcut for `Option.of(optional.orElse(null))`. Using `of(Optional)` would raise ambiguities regarding `of(T)`. Let's call it `ofOptional(Optional)`.